stingray/fourier.py

```diff
@@ -0,0 +1,1286 @@
+        Total number of counts in the light curve
+    norm : str, default "abs"
+        Normalization of the periodogram. One of ["abs", "frac", "leahy"]
+
+    Examples
+    --------
+    >>> poisson_level(norm="leahy")
+    2.0
+    >>> poisson_level(meanrate=10., norm="abs")
+    20.0
+    >>> poisson_level(meanrate=10., norm="frac")
+    0.2
+    >>> poisson_level(Nph=10, norm="none")
+    10.0
+    >>> poisson_level(meanrate=10., norm="asdfwrqfasdh3r")
+    Traceback (most recent call last):
+    ...
+    ValueError: Unknown value for norm: asdfwrqfasdh3r...
+    >>> poisson_level(meanrate=10, norm="none")
+    Traceback (most recent call last):
+    ...
+    ValueError: Bad input parameters for norm none...
+    >>> poisson_level(Nph=10, norm="abs")
+    Traceback (most recent call last):
+    ...
+    ValueError: Bad input parameters for norm abs...
+    """
+    bad_input = norm.lower() in ["abs", "frac"] and meanrate is None
+    bad_input = bad_input or (norm.lower() == "none" and Nph is None)
+
+    if bad_input:
+        raise ValueError(f"Bad input parameters for norm {norm}: Nph={Nph}, meanrate={meanrate}")
+
+    if norm == "abs":
+        return 2. * meanrate
+    if norm == "frac":
+        return 2. / meanrate
+    if norm == "leahy":
+        return 2.0
+    if norm == "none":
+        return float(Nph)
+    raise ValueError(f"Unknown value for norm: {norm}")
+
+
+def normalize_frac(power, dt, N, mean):
+    """Fractional rms normalization, from the variance of the lc.
+
+    Examples
+    --------
+    >>> mean = var = 1000000
+    >>> N = 1000000
+    >>> dt = 0.2
+    >>> meanrate = mean / dt
+    >>> lc = np.random.poisson(mean, N)
+    >>> pds = np.abs(fft(lc))**2
+    >>> pdsnorm = normalize_frac(pds, dt, lc.size, mean)
+    >>> np.isclose(pdsnorm[1:N//2].mean(), poisson_level(meanrate=meanrate,norm="frac"), rtol=0.01)
+    True
+    """
+    #     (mean * N) / (mean /dt) = N * dt
+    #     It's Leahy / meanrate;
+    #     Nph = mean * N
+    #     meanrate = mean / dt
+    #     norm = 2 / (Nph * meanrate) = 2 * dt / (mean**2 * N)
+
+    return power * 2. * dt / (mean ** 2 * N)
+
+
+def normalize_abs(power, dt, N):
+    """Absolute rms normalization, from the variance of the lc.
+
+    Examples
+    --------
+    >>> mean = var = 100000
+    >>> N = 1000000
+    >>> dt = 0.2
+    >>> meanrate = mean / dt
+    >>> lc = np.random.poisson(mean, N)
+    >>> pds = np.abs(fft(lc))**2
+    >>> pdsnorm = normalize_abs(pds, dt, lc.size)
+    >>> np.isclose(pdsnorm[1:N//2].mean(), poisson_level(meanrate=meanrate, norm="abs"), rtol=0.01)
+    True
+    """
+    #     It's frac * meanrate**2; Leahy / meanrate * meanrate**2
+    #     Nph = mean * N
+    #     meanrate = mean / dt
+    #     norm = 2 / (Nph * meanrate) * meanrate**2 = 2 * dt / (mean**2 * N) * mean**2 / dt**2
+
+    return power * 2. / N / dt
+
+
+def normalize_leahy_from_variance(power, variance, N):
+    """Leahy+83 normalization, from the variance of the lc.
```

Comment:
I'm curious what the use case is for two different formulations for the Leahy normalization, one from the variance and one from the total number of photons? 

Also, I'm going to go to emojis again: 📜 📜 📜

Reply:
This one also works with non-Poisson data!

Reply:
No, the flux might be background subtracted!